src/pyhf/readxml.py

```diff
@@ -22,12 +22,12 @@ def extract_error(h):
     bin uncertainties are then Poisson, and so the `sqrt(entries)`.
 
     Args:
-        h (uproot.rootio.TH1 object): The histogram
+        hist (uproot.rootio.TH1 object): The histogram
 
     Returns:
         list: The uncertainty for each bin in the histogram
     """
-    err = h.variances if h.variances.any() else h.numpy()[0]
+    err = hist.variances if hist.variances.any() else hist.to_numpy()[0]
```

Comment:
@jpivarski This will raise the error

```
       err = hist.variances if hist.variances.any() else hist.to_numpy()[0]
E       AttributeError: 'Model_TH1F_v2' object has no attribute 'variances'
```

in tests, but doing a quick search through the `uproot4` codebase I don't see `variances`. Can you advise? I'll look again after I'm out of meetings.

Reply:
I think https://github.com/scikit-hep/uproot/issues/511 might be what I was looking for. I'll read that later to see what the API decision was.

Reply:
That is where it's being discussed. It's in flux because we want to get all the libraries to have the same interface. Currently, there's `values_errors()`, which will give you a 2-tuple of values and errors, which you'd have to square to get variances. IN this case, it looks like you're asking if `fSumw2` exists. For that, you might want to

```python
>>> f["hpx"].member("fSumw2", none_if_missing=True)
<TArrayD [] at 0x78f865ba4450>
```

because then you can

```python
>>> if f["hpx"].member("fSumw2", none_if_missing=True): print("exists")
... 
```

The big picture is that the many methods TH* used to have are being removed in favor of a slimmer interface that mostly converts, so that people use real histogram libraries, rather than the few methods that had been added to Uproot.

Reply:
Thanks @jpivarski. @kratsg @lukasheinrich we should join the discussion over at https://github.com/scikit-hep/uproot/issues/511